### DISCUSSION TITLE
Possible to prompt for "Add missing function declaration" in SFC template?

### DISCUSSION THREAD

**Discussion by CharlieDigital** [vuejs/language-tools#1901] [Q&A] [answered]
I'm not sure if I've mis-configured something or this is a feature that is simply not yet supported.

When I bind an `@click` in an SFC template with a function that does not yet exist, VS Code does not give me a prompt to **Add missing function declaration**:

![vue-vscode](https://user-images.githubusercontent.com/779970/192005803-3399ea6a-b447-415a-a7d8-8ced95737baf.gif)

Using Vue in TSX, this works as expected and I get code actions:

![vue-prompt](https://user-images.githubusercontent.com/779970/192006245-cad21ebb-94e2-4f9c-8d90-dc32cb4c1776.gif)

Use case is that I will:

1. bind an event to a function name
2. generate the declaration
3. click through to jump to the line and start writing the handler

- I have switched to take over mode.
- Everything else works as expected.

**Comment by johnsoncodehk** [comment, marked as answer]
TypeScript's original behavior does not directly support this use case, it is possible to implement it in special ways, but there are currently no plans to do so.
